common/nym-lp/src/state_machine.rs

```diff
@@ -170,35 +213,147 @@ impl LpStateMachine {
+                                "prepare_kkt_request returned None for initiator".to_string(),
                             );
                             let reason = err.to_string();
                             result_action = Some(Err(err));
                             LpState::Closed { reason }
                         }
                     }
                 } else {
-                    // Responder waits for the first message, transition to Handshaking to wait.
-                    LpState::Handshaking { session }
+                    // Responder waits for KKT request
+                    LpState::KKTExchange { session }
                     // No action needed yet, result_action remains None.
                 }
             }
 
+            // --- KKTExchange State ---
+            (LpState::KKTExchange { session }, LpInput::ReceivePacket(packet)) => {
+                // Check if packet lp_id matches our session
+                if packet.header.session_id() != session.id() {
+                    result_action = Some(Err(LpError::UnknownSessionId(packet.header.session_id())));
+                    LpState::KKTExchange { session }
+                } else {
+                    use crate::message::LpMessage;
+
+                    // Packet message is already parsed, match on it directly
+                    match &packet.message {
+                        LpMessage::KKTRequest(kkt_request) if !session.is_initiator() => {
+                            // Responder processes KKT request
+                            // Convert X25519 public key to KEM format for KKT response
+                            use nym_kkt::ciphersuite::EncapsulationKey;
+
+                            // Get local X25519 public key by deriving from private key
+                            let local_x25519_public = session.local_x25519_public();
+
+                            // Convert to libcrux KEM public key
+                            match libcrux_kem::PublicKey::decode(
+                                libcrux_kem::Algorithm::X25519,
+                                local_x25519_public.as_bytes(),
+                            ) {
+                                Ok(libcrux_public_key) => {
+                                    let responder_kem_pk = EncapsulationKey::X25519(libcrux_public_key);
+
+                                    match session.process_kkt_request(&kkt_request.0, &responder_kem_pk) {
+                                        Ok(kkt_response_message) => {
+                                            match session.next_packet(kkt_response_message) {
+                                                Ok(response_packet) => {
+                                                    result_action = Some(Ok(LpAction::SendPacket(response_packet)));
+                                                    // After KKT exchange, move to Handshaking
+                                                    LpState::Handshaking { session }
```

Comment:
so many different levels of indentation makes it quite hard to read - could we perhaps split up this function? maybe separate handler per state?